### PR TITLE
Avoid restarts from yum install workaround for resolved issue 20608

### PR DESCRIPTION
**Pull request by kcd83**
Would you please consider merging this as the bug was fixed a few versions ago. The workaround triggered datadog-agent to restart on every run and so I have fixed that too.

Thanks

### CODE REVIEW

**Comment by kcd83** [comment]
#147

(Playing with how github references PRs from issues)

**Comment by kcd83** [comment]
@olivielpeau could you please take a look at this?

File: tasks/pkg-redhat.yml [hunk 1]
```diff
@@ -41,7 +41,21 @@
...
 # https://github.com/ansible/ansible/issues/20608
-- name: Install latest datadog-agent package
+- name: Install latest datadog-agent package issue 20608 workaround
   command: yum install -y datadog-agent
-  when: datadog_agent_version == ""
+  register: yum_install_result
+  changed_when: ( yum_install_result.stdout_lines | last ) != "Nothing to do"
+  when:
+  - datadog_agent_version == ""
+  - ansible_version is version_compare(2.3, '<')
```
Comment (dabcoder) [inline tasks/pkg-redhat.yml]: We actually only support >= 2.4: https://github.com/DataDog/ansible-datadog/blob/ec5e5fd736982e87d4a31f06a8ba484f97ab8340/meta/main.yml#L6 (and version 2.3 has been marked as EOL: https://docs.ansible.com/ansible/2.4/release_and_maintenance.html#release-status) so we can remove that case and keep the one below given that the `#20608` issue is now resolved in >= 2.4.

File: tasks/pkg-redhat.yml [hunk 2]
```diff
@@ -41,7 +41,21 @@
...
   notify: restart datadog-agent
+
+- name: Install latest datadog-agent package
+  yum:
+    name: datadog-agent
+    update_cache: yes
+    state: latest
+  when:
+  - datadog_agent_version == ""
+  - ansible_version is version_compare(2.3, '>=')
```
Comment (dabcoder) [inline tasks/pkg-redhat.yml]: Given that this role support >= 2.4 we can drop this condition.

**Review by dabcoder — CHANGES_REQUESTED** [review]
Left a couple of notes, thanks!

**Comment by dabcoder** [comment]
@kcd83 Did you have a chance to review my comments above?

File: tasks/pkg-redhat.yml (hunk 1 above)
Comment (kcd83) [reply tasks/pkg-redhat.yml]: Good idea, much simpler

File: tasks/pkg-redhat.yml [hunk 3]
```diff
@@ -39,9 +39,11 @@
...
-# https://github.com/ansible/ansible/issues/20608
 - name: Install latest datadog-agent package
-  command: yum install -y datadog-agent
-  when: datadog_agent_version == ""
+  yum:
+    name: datadog-agent
+    update_cache: yes
+    state: latest
+  when:
+  - datadog_agent_version == ""
```
Comment (kcd83) [inline tasks/pkg-redhat.yml:48]: This could got back to one line, and do you want `not ansible_check_mode` to be consistent with above

**Comment by kcd83** [comment]
Good points, yes I finally have a moment to check this

File: tasks/pkg-redhat.yml (hunk 3 above)
Comment (dabcoder) [reply tasks/pkg-redhat.yml:48]: Good point, we do it for Debian based distros: https://github.com/DataDog/ansible-datadog/blob/34e4275e079d71b3ea775ad2ac3d73e2361dd0cc/tasks/pkg-debian.yml#L61 but not for Red Hat nor Suse. We could add for both. I believe your branch does not include the `pkg-suse.yaml` file so you'll need to rebase on master to make that change (if you want?).

Thanks!

**Comment by dabcoder** [comment]
Took care of that in https://github.com/DataDog/ansible-datadog/pull/199, but thanks for your work here @kcd83, much appreciated!

**Comment by kcd83** [comment]
Thanks @dabcoder !

**Pull request closed without merging**